io-sim/src/Data/Deque/Strict.hs

```diff
@@ -0,0 +1,41 @@
+{-# LANGUAGE LambdaCase #-}
+
+-- | A minimal implementation of a strict deque.
+--
+module Data.Deque.Strict where
+
+import           Prelude hiding (head, init, tail)
+import           Data.Foldable (foldl')
+import qualified Data.List as List
+
+data Deque a = Deque ![a] ![a]
+
+instance Semigroup (Deque a) where
+    Deque as bs <> Deque as' bs' =
+      Deque as (bs ++ reverse as' ++ bs')
```

Comment:
Shouldn't this be

```suggestion
      Deque as (bs' ++ reverse as' ++ bs)
```

Reply:
You're right, I messed this up after trying to do that on the head side.